R/dist_quantiles.R

```diff
@@ -5,6 +5,9 @@ new_quantiles <- function(values = double(), quantile_levels = double()) {
   vec_cast(values, double())
   vec_cast(quantile_levels, double())
   stopifnot(length(values) == length(quantile_levels))
```

Comment:
```suggestion
  stopifnot(length(values) == length(quantile_levels))
  values <- unname(values)
```
this can prevent `vctrs::vec_cast_common` from producing things like
```
Error:
! Can't convert `..1` <dist_quantiles> to <dist_quantiles>
```
but you'll need to accept/review the snapshot tests.

Reply:
Will do.